README.md

```diff
@@ -99,30 +99,33 @@ Contact: firstname.lastname@hpi.de
 
 ## Contributors
 
--	Yannick  Bäumer
```

Comment:
Could you explain why LZ4 suffers from the data being sorted?

Reply:
I was wondering the same. @cmfcmf: which commit did you benchmark?
I’d guess the current master as it shows the relatively good decompression speed for a full sequential operation (both not optimized paths) while the sorted case uses the PointAccessIterator to extract single values during binary search (which is tackled in a current PR).
Does that guess make sense?

Reply:
Ah sure, that's probably it. For the future, I would strongly suggest to not put out graphs that show such obvious discrepancies without commenting on those.

Reply:
> I’d guess the current master as it shows the relatively good decompression speed for a full sequential operation (both not optimized paths) while the sorted case uses the PointAccessIterator to extract single values during binary search (which is tackled in a current PR).

No, neither uses PointAccessIterators. These would only be used for ReferenceSegments in combination with PosLists and position_filters. The name `PointAccessIterator` is a bit misleading, IMHO, because neither `PointAccessIterator` nor the normal iterators allow random access.

I've tried to track down why this happens using Valgrind, but wasn't able to find the root cause just yet.

Here's a comparison of the `_scan_non_reference_segment` method for sorted and sorted, but not marked as such, LZ4Segments. As you see, compression takes the same amount of time for both cases.

### Sorted
![grafik](https://user-images.githubusercontent.com/2145092/53962599-c5b55b80-40eb-11e9-92ee-0e17ccaf3070.png)

### SortedNotMarked
![grafik](https://user-images.githubusercontent.com/2145092/53962562-b3d3b880-40eb-11e9-856d-08914f3fdb56.png)




I'm still investigating what exactly is happening and why the AnySegmentIterator doesn't use `advance()` but only `increment()`.

Reply:
I'm not 100% sure if its related, but currently SegmentIterators aren't proper LegacyRandomAccessIterators: https://github.com/hyrise/hyrise/issues/1531